core/evancore.vue.libs/src/components/permissions-editor/permissions-editor.ts

```diff
@@ -182,28 +203,36 @@ class PermissionsEditor extends mixins(EvanComponent) {
     this.isLoading = true;
     this.setUserNameWithAddress();
     this.containersPermissions = null;
-    this.containersPermissions = await this.loadPermissions(this.selectedContact)
-      .catch((e: Error) => {
-        console.log('Error loading permissions', e.message);
-        this.isLoading = false;
-      });
+    this.containersPermissions = await this.loadPermissions(
+      this.selectedContact,
+    ).catch((e: Error) => {
+      console.error('Error loading permissions', e.message);
+      this.isLoading = false;
+    });
 
-    this.initialPermissions = this.containersPermissions ? clone(this.containersPermissions) : this.initialPermissions;
+    this.initialPermissions = this.containersPermissions
+      ? clone(this.containersPermissions)
+      : this.initialPermissions;
 
     this.isLoading = false;
   }
 
   /**
    * Calls the `updatePermissions` function from properties with the updated containersPermissions object.
    */
-  async writePermissions() {
+  async writePermissions(): Promise<void> {
     this.isLoading = true;
 
     try {
-      await this.updatePermissions(this.runtime, this.selectedContact, this.containersPermissions, this.initialPermissions);
+      await this.updatePermissions(
+        this.runtime,
+        this.selectedContact,
+        this.containersPermissions,
+        this.initialPermissions,
+      );
       this.initialPermissions = clone(this.containersPermissions);
     } catch (ex) {
-      return console.error('Error writing permissions', ex.message);
+      console.error('Error writing permissions', ex.message);
```

Comment:
After this catch block, `cancel` function will be called, that will close the sidepanel. Is this intended?

Reply:
Good question. Didn't even consider it. Both options (just infinite loading while displaying no UI error and just closing it) suck kind of.